src/platform/plugins/shared/cps/server/plugin.ts

```diff
@@ -8,33 +8,48 @@
  */
 
 import type { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from '@kbn/core/server';
+import type { Observable } from 'rxjs';
+import { firstValueFrom } from 'rxjs';
 import { registerRoutes } from './routes';
+import type { CPSConfig } from './config';
+import type { CPSServerSetup } from './types';
 
-// No setup contract needed for this plugin
-
-export class CPSServerPlugin implements Plugin<{}> {
+export class CPSServerPlugin implements Plugin<CPSServerSetup> {
   private readonly initContext: PluginInitializerContext;
   private readonly isServerless: boolean;
+  private readonly config$: Observable<CPSConfig>;
 
   constructor(initContext: PluginInitializerContext) {
     this.initContext = { ...initContext };
     this.isServerless = initContext.env.packageInfo.buildFlavor === 'serverless';
+    this.config$ = initContext.config.create();
   }
 
-  public setup(core: CoreSetup, plugins: {}) {
-    const { initContext } = this;
+  public setup(core: CoreSetup) {
+    void this.setCpsFeatureFlagAsync(core);
+    const { initContext, config$ } = this;
 
     // Register route only for serverless
     if (this.isServerless) {
       registerRoutes(core, initContext);
     }
 
-    return {};
+    return {
+      getCpsEnabled: async () => {
+        const { cpsEnabled } = await firstValueFrom(config$);
+        return cpsEnabled;
+      },
+    };
   }
 
   public start(core: CoreStart) {
     return {};
   }
 
   public stop() {}
+
+  private async setCpsFeatureFlagAsync(core: CoreSetup) {
```

Comment:
I couldn't get this to work from another plugin without changing the consuming plugin's setup contract to `async`.